### PR TITLE
Move public IP address detection out of bash

### PR DESCRIPTION
**Pull request by mvines**
Includes further refinement/churn of the `setup.sh/`solana-fullnode-config` CLI arguments.   The options are:

Use local address (may be a private IP):
```sh
$ ./multinode-demo/setup.sh
.. or ..
$ ./multinode-demo/setup.sh -l
```

Use the detected public address (forced for the snap-based fullnode daemon):
```sh
./multinode-demo/setup.sh -p
```

### CODE REVIEW

**Comment by mvines** [comment]
@garious I think this is a little nicer for the snap setup at least.   My Rust is still pretty terrible so donno what you think about bringing in that `curl` create dependency

File: src/nat.rs [hunk 1]
```diff
@@ -16,6 +19,31 @@ pub struct UdpSocketPair {
     pub sender: UdpSocket,   // Locally bound socket to send via public address
 }
 
+/// Tries to determine the public IP address of this machine, returns 0.0.0.0 on
+/// failure
+pub fn get_public_ip_addr() -> IpAddr {
```
Comment (garious) [inline src/nat.rs]: This should return a Result. There's enough things to go wrong.

**Comment by garious** [comment]
Good thinking on defaulting local.

**Comment by mvines** [comment]
Gross this brings in an openssl dependency that's breaking `sagan` and the snap build.

**Comment by garious** [comment]
An openssl dep seems pretty unavoidable. A broken one is disappointing.

**Comment by garious** [comment]
nat put a dependency on hyper. Might as well add https://docs.rs/reqwest/0.8.6/reqwest/?

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to merge conflict

**Comment by mvines** [comment]
🚍

**Pull request merged**